### PR TITLE
Fixed setup.py to handle CUDNN_LIBRARY envvar with aten

### PR DESCRIPTION
**Pull request by Erotemic**
I ran into an issue building from source. When I ran `python setup.py install` things seemed to build ok, but at runtime it said that the cudnn that I build with was different than the one I was linking against. Sure enough I thought I was building with 7, but I was linking against 6. 

It turns out the issue was that `aten` looks at the `CUDNN_LIBRARY` variable, but this was not being set in `setup.py`. Only the `CUDNN_LIB_DIR` was handled. 

This change modifies `tools/setup_helpers/cudnn.py` to look for the `CUDNN_LIBRARY` variable and checks to make sure it agrees with `CUDNN_LIB_DIR`. I then appropriately pass down the variable to `aten` in `torch/lib/build_libs.sh`. 

Finally during my debugging I added an `if __name__ == '__main__'` to setup.py to avoid calling setup when I imported the file to debug its variables. It seems like it should be there so I left it in.

### CODE REVIEW

**Review by soumith — APPROVED** [review]
looks good to me. thanks for debugging this and fixing things.

File: setup.py [hunk 1]
```diff
@@ -715,18 +717,19 @@ def make_relative_rpath(path):
...
-          'lib/*.so*', 'lib/*.dylib*', 'lib/*.dll', 'lib/*.lib',
-          'lib/torch_shm_manager',
-          'lib/*.h',
-          'lib/include/TH/*.h', 'lib/include/TH/generic/*.h',
-          'lib/include/THC/*.h', 'lib/include/THC/generic/*.h',
-          'lib/include/ATen/*.h',
-      ]},
-      install_requires=['pyyaml', 'numpy'],
-      )
+if __name__ == '__main__':
```
Comment (apaszke) [inline setup.py]: Why the `if`? I don't think we ever want to import this file anywhere

**Review by apaszke — COMMENTED** [review]
Nice catch, thanks for the PR!

File: setup.py (hunk 1 above)
Comment (soumith) [reply setup.py]: as he mentioned, it's to debug. if we want to know variable values in setup.py, we can import it in an interpreter and inspect (in that case the build is not run).

**Pull request merged**